src/Swarm/Game/Step.hs

```diff
@@ -1920,11 +1942,25 @@ execConst c vs s k = do
       ]
 
   rectCells :: Integer -> Integer -> Integer -> Integer -> [V2 Int32]
-  rectCells x1 y1 x2 y2 = [V2 x y | x <- [fromIntegral xMin .. fromIntegral xMax], y <- [fromIntegral yMin .. fromIntegral yMax]]
+  rectCells x1 y1 x2 y2 =
+    rectCellsInt32
+      (fromIntegral x1)
+      (fromIntegral y1)
+      (fromIntegral x2)
+      (fromIntegral y2)
+
+  rectCellsInt32 :: Int32 -> Int32 -> Int32 -> Int32 -> [V2 Int32]
+  rectCellsInt32 x1 y1 x2 y2 = [V2 x y | x <- [xMin .. xMax], y <- [yMin .. yMax]]
    where
     (xMin, xMax) = sortPair (x1, x2)
     (yMin, yMax) = sortPair (y1, y2)
 
+  hasSightLineTo :: HasRobotStepState sig m => Location -> Location -> m Bool
```

Comment:
Maybe add a comment that it only makes sense to use this with two locations that share the same x- or y-coordinate?